AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -639,62 +646,93 @@ - (BOOL)isLayerBacked
   return _flags.layerBacked;
 }
 
-#pragma mark - Layout measurement calculation
+#pragma mark - Layout measurement and sizing
+
+- (ASSize)size
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  return _size;
+}
+
+- (void)setSize:(ASSize)size
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  if (ASSizeEqualToSize(_size, size) == NO) {
+    _size = size;
+    [self invalidateCalculatedLayout];
+  }
+}
+
+ASLayoutableSizeForwarding
+ASLayoutableSizeHelperForwarding
 
+// Deprecated
 - (CGSize)measure:(CGSize)constrainedSize
 {
-  return [self measureWithSizeRange:ASSizeRangeMake(CGSizeZero, constrainedSize)].size;
+    return [self layoutThatFits:ASSizeRangeMake(CGSizeZero, constrainedSize)].size;
+}
+
+- (ASLayout *)layoutThatFits:(ASSizeRange)constrainedSize
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  return [self measureWithSizeRange:constrainedSize];
+#pragma clang diagnostic pop
 }
 
+// Deprecated
 - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
 {
+  return [self layoutThatFits:constrainedSize parentSize:constrainedSize.max];
```

Comment:
Because we call `measureWithSizeRange:` in `layoutThatFits:` we have to call `layoutThatFits:parentSize:` in this place